src/Core/Framework/Plugin/Api/PluginController.php

```diff
@@ -0,0 +1,163 @@
+<?php declare(strict_types=1);
+
+namespace TeamDeployment\Plugin\Deployment\Core\Framework\Plugin\Api;
+
+use Composer\IO\NullIO;
+use Shopware\Core\Framework\Context;
+use Shopware\Core\Framework\DataAbstractionLayer\EntityRepositoryInterface;
+use Shopware\Core\Framework\DataAbstractionLayer\Search\Criteria;
+use Shopware\Core\Framework\DataAbstractionLayer\Search\EntitySearchResult;
+use Shopware\Core\Framework\Plugin\Exception\PluginComposerJsonInvalidException;
+use Shopware\Core\Framework\Plugin\PluginLifecycleService;
+use Shopware\Core\Framework\Plugin\PluginService;
+use Shopware\Core\Framework\Plugin\Requirement\Exception\RequirementStackException;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\Routing\Annotation\Route;
+
+/**
+ * @author Amir El Sayed <a.elsayed@basecom.de>
+ */
+class PluginController extends AbstractController
+{
+    /**
+     * @var PluginService
+     */
+    private $pluginService;
+
+    /**
+     * @var PluginLifecycleService
+     */
+    private $pluginLifecycleService;
+
+    /**
+     * @var EntityRepositoryInterface
+     */
+    private $pluginRepo;
+
+    public function __construct(
+        PluginService $pluginService,
+        PluginLifecycleService $pluginLifecycleService,
+        EntityRepositoryInterface $pluginRepo
+    ) {
+        $this->pluginService = $pluginService;
+        $this->pluginLifecycleService = $pluginLifecycleService;
+        $this->pluginRepo = $pluginRepo;
+    }
+
+    /**
+     * @Route("/api/v{version}/_action/plugin/deploy", name="api.action.plugin.installAll", methods={"POST"})
+     * @param Context $context
+     *
+     * @return JsonResponse
+     * @throws \Shopware\Core\Framework\DataAbstractionLayer\Exception\InconsistentCriteriaIdsException
+     * @throws \Shopware\Core\Framework\Plugin\Exception\PluginComposerJsonInvalidException
+     * @throws \Shopware\Core\Framework\Plugin\Exception\PluginNotInstalledException
+     * @throws \Shopware\Core\Framework\Plugin\Requirement\Exception\RequirementStackException
+     */
+    public function deployPlugins(Context $context): JsonResponse
+    {
+        $pluginList = $this->refreshPlugins($context);
+
+        foreach($pluginList as $plugin) {
+            $this->pluginLifecycleService->updatePlugin($plugin, $context);
```

Comment:
How would this behave for plugins which are not installed yet?

Maybe check the state of each plugin before deciding which action to take?